File: java/simple_guess/NumGuess.java
import java.util.Random;
import java.util.Scanner;  
  
public class NumGuess {

    static boolean compareVals(int guess, int answer) {  
        System.out.println("You guessed: " + guess);  
  
        if (guess == answer) {  
            System.out.println("Congrats, Your answer is Correct!");  
            return true;  
        } else {  
            if (guess > answer) {  
                System.out.println("Guess too big!");  
            } else {  
                System.out.println("Guess too small!");  
            }  
            return false;  
        }  
    } 

    public static void main(String[] args) {  
        System.out.println("Guess the number!");  
  
        int maxTries = 3;
        int counter = maxTries;  
        Random rand = new Random();  
        int answer = rand.nextInt(10);  
  
        Scanner scanner = new Scanner(System.in);  
  
        while (true) {  
            if (counter == 0) {  
                System.out.println("Sorry, you have reached the maximum tries of " + maxTries + " times.");  
                return;  
            }  
  
            System.out.print("Please enter your guess.(below 10): ");  
            String guess = scanner.next();  
  
            try {  
                int guessInt = Integer.parseInt(guess);  
  
                if (guessInt < 11) {  
                    // System.out.println("You entered: " + guess + ", the correct is " + answer); 
                    if (compareVals(guessInt, answer)) {  
                        scanner.close();
                        return;  
                    }  
                } else {  
                    System.out.println("Please enter a number below 10!");  
                    counter--;  
                }  
            } catch (NumberFormatException e) {  
                System.err.println("Please type a valid integer!");  
                counter--;  
            }  
        }  

    }  
}  

File: elixir/num_guess/lib/num_guess.ex
defmodule NumGuess do
  def process_number() do

    IO.puts("Guess the number!")

    max_tries = 3
    answer = :rand.uniform(10);
    get_input(max_tries,answer)

  end

  # recursive loop as elixir has no loops
  defp get_input(tries,answer) do

    if tries == 0 do
      IO.puts("Sorry, you have reached the max tries of 3 times.")
      :ok
    else
      guess = IO.gets("Please input a number. (below 10) : ");

      if Integer.parse(String.trim(guess)) == :error do
        IO.puts("The input is not a valid integer")
        get_input(tries - 1,answer)

      else

        guessInt = guess |> String.trim |> String.to_integer
        if 0 < guessInt and guessInt< 11 do
          if not compare_vals(guessInt,answer) do
            get_input(tries,answer)
          end
        else
          IO.puts("Please enter a number below 10")
          get_input(tries - 1,answer)
        end

      end

    end

  end

  # compare guess to real
  defp compare_vals(guessInt,answer) do
    # IO.puts(["Correct number is #{correct}"])
    IO.puts("Your guess is #{guessInt}")
    # IO.puts("You guessed: #{guessInt}, the correct is #{correct}.")

    case {guessInt, answer} do
      {guess, guess} ->
        IO.puts("Your answer is correct")
        true
      {guess, answer} when guess > answer ->
        IO.puts("Guess too big!")
        false
      _ ->
        IO.puts("Guess too small!")
        false
    end

  end

end


# NumGuess.process_number()
